eta/core/datasets.py

```diff
@@ -2751,20 +2751,21 @@ class SchemaFilter(DatasetTransformer):
     is None, no filtering is done.
     '''
 
-    def __init__(
-            self, schema, remove_objects_without_attrs=False,
-            prune_empty=True):
-        '''Initialize the SchemaFilter with a schema.
+    def __init__(self, schema, object_labels_to_filter=None, prune_empty=True):
```

Comment:
With this refactor you'd have to set `object_labels_to_filter = []` if you wanted to NOT filter objects with no attributes. That's a bit confusing. I think you still need the `remove_objects_without_attrs` flag here to more intuitively control whether you want filtering to happen

Reply:
True, I added another boolean param to help clarify thtis